Userland/Libraries/LibUnicode/CharacterTypes.cpp

```diff
@@ -222,6 +222,52 @@ u32 to_unicode_uppercase(u32 code_point)
 #endif
 }
 
+Optional<StringView> code_point_name([[maybe_unused]] u32 code_point)
+{
+#if ENABLE_UNICODE_DATA
+    auto name = Detail::code_point_name(code_point);
+    if (name.is_null())
+        return {};
+    return name;
+#else
+    return {};
+#endif
+}
+
+Optional<StringView> code_point_unicode_1_name([[maybe_unused]] u32 code_point)
+{
+#if ENABLE_UNICODE_DATA
+    auto name = Detail::code_point_unicode_1_name(code_point);
+    if (name.is_null())
+        return {};
+    return name;
+#else
+    return {};
+#endif
+}
+
+Optional<StringView> code_point_useful_name([[maybe_unused]] u32 code_point)
+{
+#if ENABLE_UNICODE_DATA
+    auto name = Detail::code_point_name(code_point);
+
+    // If even Unicode 2 has no name for it, there is no hope.
+    if (name.is_null())
+        return {};
+    if (name != "<control>")
+        return name;
+
+    // If Unicode 2 calls it "<control>", the Unicode 1 name is actually more useful, if it exists.
```

Comment:
IMO the logic here could be done by the generator. The only caller wants the "useful" name, therefore let's just not generate non-useful names until we need them.

Are there any large amounts of repeated names? If so we may want to consider a master string table analogous to f9e605397c13730bd1382b17069790e398bf428a to keep the side of libunicode.so smaller. If there's not many repetitions, then no worries.

Reply:
The only repeated names are the empty string, and a handful of times it's `<control>`. The empty strings are represented by null StringViews, and collapsing the `<control>`s doesn't seem worth the effort; I'll think about it, though.

Hmm, maybe the generated code could look something like:
```
static const StringView s_literal_control = "<control>"sv;
… // and then in the array:
    { 9, s_literal_control, "CHARACTER TABULATION"sv },
```

Obviously, that only works if the compiler doesn't already do that. I'll experiment a bit.

Reply:
> IMO the logic here could be done by the generator. The only caller wants the "useful" name, therefore let's just not generate non-useful names until we need them.

Given that the library is called `LibUnicode`, that each codepoint has a clearly defined "name" according to spec, and that our interpretation of "useful" deviates from that, I disagree. We would need to actively throw data away, which would bite us in the backend if we ever expose it to JavaScript or similar.

Reply:
That doesn't really follow. The three LibUnicode generators actively discard, and even mutate, data from the UCD and CLDR to produce what we need. You don't need to call it exactly `name`, you can call it `display_name` or something to deviate from the spec's terminology.

But we should always generate the least amount of data we need for the task at hand. Keeping dead data around just in case Ecma or anyone adds an API for it in the future is an anti-goal of these generators.

Reply:
I went with the `display_name` suggestion. Thanks! :D